tensorflow_datasets/core/folder_dataset/image_folder.py

```diff
@@ -163,20 +171,33 @@ def _as_dataset(
     if shuffle_files:
       ds = ds.shuffle(len(examples))
     ds = ds.map(_load_example, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+    if not decoders:
+      decode_fn = _default_decode_fn
+    else:
+      decode_fn = functools.partial(
```

Comment:
Would there be a drawback about always decoding through `self.info.features.decode_example` even when `decoders` is `None` ?

Reply:
`decode_example` in tensorflow_datasets/core/features/image_feature.py does expand_animation=True and does not hardcode channels=3, which is different from the current behavior as far as I can tell.